### PR TITLE
PIM-4413: Create a filter for a product to import only updated values

### PR DESCRIPTION
**Pull request by skeleton**
| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | N
| BC breaks?           | N
| CI currently passes? | ?
| Tests pass?          | Y 
| Scenarios pass?      | Y
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | N
| Fixed tickets        | PIM-4413
| DB schema updated?   | N
| Migration script?    | N
| Doc PR               | N

### CODE REVIEW

**Comment by phaseinducer** [comment]
I think "ProductFilter" is a too generic name for its purpose. Plus it's confusing since other stuff in the PIM are also named *Filter and are really different from that.

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/GroupTypeRepository.php [hunk 1]
```diff
@@ -50,12 +51,31 @@ public function createDatagridQueryBuilder()
         return $qb;
     }
 
+
```
Comment (nidup) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/GroupTypeRepository.php]: extra line

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/GroupTypeRepository.php [hunk 2]
```diff
@@ -50,12 +51,31 @@ public function createDatagridQueryBuilder()
...
+    {
+        try {
+            return $this->createQueryBuilder('group_type')
+                ->innerJoin('group_type.groups', 'g')
+                ->select('group_type.variant')
+                ->where('g.code = :code')
+                ->setParameter('code', $code)
+                ->getQuery()
+                ->getSingleScalarResult();
+        } catch (UnexpectedResultException $e) {
```
Comment (nidup) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/GroupTypeRepository.php]: when this UnexpectedResultException can happens?

Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/GroupTypeRepository.php]: When a group does not exist

**Comment by nidup** [comment]
missing specs?

File: src/Pim/Component/Connector/ArrayConverter/Flat/Product/FieldConverter.php [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 namespace Pim\Component\Connector\ArrayConverter\Flat\Product;
 
+use Pim\Bundle\CatalogBundle\Repository\GroupTypeRepositoryInterface;
 use Pim\Component\Connector\ArrayConverter\Flat\Product\Splitter\FieldSplitter;
```
Comment (nidup) [inline src/Pim/Component/Connector/ArrayConverter/Flat/Product/FieldConverter.php]: you should rebase  use Pim\Component\Connector\ArrayConverter\Flat\Product\Splitter\FieldSplitter has been move to Pim\Component\Connector\ArrayConverter\Flat\Product\FieldSplitter;

**Comment by phaseinducer** [comment]
GTM

**Pull request merged**